### PR TITLE
Fix golint failures in pkg/credentialprovider

### PR DESCRIPTION
**Pull request by zxh326**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

This fixes golint errors in the following file:

```
Errors from golint:
pkg/credentialprovider/config.go:41:6: type DockerConfigJson should be DockerConfigJSON
pkg/credentialprovider/config.go:44:2: struct field HttpHeaders should be HTTPHeaders
pkg/credentialprovider/config.go:52:6: exported type DockerConfigEntry should have comment or be unexported
pkg/credentialprovider/config.go:65:2: var homeJsonDirPath should be homeJSONDirPath
pkg/credentialprovider/config.go:66:2: var rootJsonDirPath should be rootJSONDirPath
pkg/credentialprovider/config.go:69:2: var configJsonFileName should be configJSONFileName
pkg/credentialprovider/config.go:72:1: exported function SetPreferredDockercfgPath should have comment or be unexported
pkg/credentialprovider/config.go:78:1: exported function GetPreferredDockercfgPath should have comment or be unexported
pkg/credentialprovider/config.go:153:6: func ReadSpecificDockerConfigJsonFile should be ReadSpecificDockerConfigJSONFile
pkg/credentialprovider/config.go:162:1: exported function ReadDockerConfigFile should have comment or be unexported
pkg/credentialprovider/config.go:171:6: type HttpError should be HTTPError
pkg/credentialprovider/config.go:173:2: struct field Url should be URL
pkg/credentialprovider/config.go:182:1: exported function ReadUrl should have comment or be unexported
pkg/credentialprovider/config.go:182:6: func ReadUrl should be ReadURL
pkg/credentialprovider/config.go:217:1: exported function ReadDockerConfigFileFromUrl should have comment or be unexported
pkg/credentialprovider/config.go:217:6: func ReadDockerConfigFileFromUrl should be ReadDockerConfigFileFromURL
pkg/credentialprovider/config.go:220:9: if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)
pkg/credentialprovider/config.go:233:6: func readDockerConfigJsonFileFromBytes should be readDockerConfigJSONFileFromBytes
pkg/credentialprovider/config.go:234:6: var cfgJson should be cfgJSON
pkg/credentialprovider/config.go:256:1: exported method DockerConfigEntry.UnmarshalJSON should have comment or be unexported
pkg/credentialprovider/config.go:275:1: exported method DockerConfigEntry.MarshalJSON should have comment or be unexported
pkg/credentialprovider/config_test.go:30:2: var configJsonFileName should be configJSONFileName
pkg/credentialprovider/config_test.go:34:2: var inputDockerconfigJsonFile should be inputDockerconfigJSONFile
pkg/credentialprovider/keyring.go:76:1: exported method BasicDockerKeyring.Add should have comment or be unexported
pkg/credentialprovider/keyring.go:162:6: func parseSchemelessUrl should be parseSchemelessURL
pkg/credentialprovider/keyring.go:162:25: func parameter schemelessUrl should be schemelessURL
pkg/credentialprovider/keyring.go:173:6: func splitUrl should be splitURL
pkg/credentialprovider/keyring.go:184:2: var globUrl should be globURL
pkg/credentialprovider/keyring.go:188:2: var targetUrl should be targetURL
pkg/credentialprovider/keyring.go:203:16: func parameter globUrl should be globURL
pkg/credentialprovider/keyring.go:203:34: func parameter targetUrl should be targetURL
pkg/credentialprovider/keyring.go:204:2: var globUrlParts should be globURLParts
pkg/credentialprovider/keyring.go:205:2: var targetUrlParts should be targetURLParts
pkg/credentialprovider/keyring.go:218:9: range var globUrlPart should be globURLPart
pkg/credentialprovider/keyring.go:219:3: var targetUrlPart should be targetURLPart
pkg/credentialprovider/keyring.go:273:6: exported type FakeKeyring should have comment or be unexported
pkg/credentialprovider/keyring.go:278:1: exported method FakeKeyring.Lookup should have comment or be unexported
pkg/credentialprovider/keyring.go:285:1: exported method UnionDockerKeyring.Lookup should have comment or be unexported
pkg/credentialprovider/keyring_test.go:28:3: struct field globUrl should be globURL
pkg/credentialprovider/keyring_test.go:29:3: struct field targetUrl should be targetURL
pkg/credentialprovider/keyring_test.go:125:3: struct field globUrl should be globURL
pkg/credentialprovider/keyring_test.go:126:3: struct field targetUrl should be targetURL
pkg/credentialprovider/keyring_test.go:233:3: struct field globUrl should be globURL
pkg/credentialprovider/keyring_test.go:234:3: struct field lookupUrl should be lookupURL


```
**Which issue(s) this PR fixes**:

Ref #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/credentialprovider/keyring.go [hunk 1]
```diff
@@ -270,18 +271,11 @@ func (dk *providersDockerKeyring) Lookup(image string) ([]AuthConfig, bool) {
 	return keyring.Lookup(image)
 }
 
-type FakeKeyring struct {
```
Comment (zxh326) [inline pkg/credentialprovider/keyring.go]: I removed this implementation because I didn't find anywhere to use it

**Comment by neolit123** [comment]
/kind cleanup
/priority backlog
/sig auth

hi, the changes seem fine but today is code freeze and we won't be able to merge the PR until code freeze is lifted:
https://github.com/kubernetes/sig-release/tree/master/releases/release-1.18

**Pull request merged**